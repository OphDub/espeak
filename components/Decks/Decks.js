import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import PropTypes from 'prop-types';

export default class Decks extends Component<Props> {
  static navigationOptions = {
    title: 'Decks',
  };

  navigateToCards(deck) {
    this.props.navigation.navigate('WordCards', deck);
  }

  render() {
    const { decks } = this.props.screenProps;

    return (
      <View style={styles.container}>
        <View style={styles.deckList}>
          {decks.map( (deck, index) => {
            const { stack_id }  = this.props.screenProps.user;
            const isActive = deck.id <= stack_id;
            const deckStyle = isActive ? styles.deck : styles.disabled;

            return (
              <TouchableOpacity
                onPress={() => this.navigateToCards(deck)}
                disabled={!isActive}
                value={deck.id}
                style={deckStyle}
                key={index} >
                <Text style={styles.deckText}>
                  {deck.category}
                </Text>
              </TouchableOpacity>
            );
          })}
        </View>
      </View>
    );
  }
}

Decks.propTypes = {
  navigation: PropTypes.object,
  screenProps: PropTypes.object
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingVertical: 40,
  },
  deckList: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  deck: {
    backgroundColor: '#3AAFB9',
    padding: 5,
    borderRadius: 8,
    margin: 10,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  disabled: {
    backgroundColor: '#B7B2B8',
    padding: 5,
    borderRadius: 8,
    margin: 10,
  },
  deckText: {
    fontSize: 20,
    color: 'white',
    textAlign: 'center',
    margin: 10,
  },
});

