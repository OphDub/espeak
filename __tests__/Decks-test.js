import React from 'react';
import { shallow, configure } from 'enzyme';
import Decks from '../components/Decks/Decks';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const word = { english: 'hi', spanish: 'hola', hint: 'o-la' };
  const wrap = renderer.create(<Decks />).toJSON();
  expect(wrap).toMatchSnapshot();
})