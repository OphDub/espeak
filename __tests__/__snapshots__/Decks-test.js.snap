// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingVertical": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  />
</View>
`;
